tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuning.scala

```diff
@@ -0,0 +1,616 @@
+    if (app.sqlTaskAggMetrics.isEmpty) { // avoid division by zero
+      maxPartitionBytesNum.toString
+    } else {
+      val taskInputSizeInMB =
+        (app.sqlTaskAggMetrics.map(_.inputBytesReadAvg).sum / app.sqlTaskAggMetrics.size) / 1024
+      if (taskInputSizeInMB > 0 &&
+        taskInputSizeInMB < MIN_PARTITION_BYTES_RANGE) {
+        // Increase partition size
+        val calculatedMaxPartitionBytes = Math.min(
+          maxPartitionBytesNum *
+            (MIN_PARTITION_BYTES_RANGE / taskInputSizeInMB),
+          MAX_PARTITION_BYTES_BOUND)
+        calculatedMaxPartitionBytes.toLong.toString
+      } else if (taskInputSizeInMB > MAX_PARTITION_BYTES_RANGE) {
+        // Decrease partition size
+        val calculatedMaxPartitionBytes = Math.min(
+          maxPartitionBytesNum /
+            (taskInputSizeInMB / MAX_PARTITION_BYTES_RANGE),
+          MAX_PARTITION_BYTES_BOUND)
+        calculatedMaxPartitionBytes.toLong.toString
+      } else {
+        // Do not recommend maxPartitionBytes
+        null
+      }
+    }
+  }
+
+  /**
+   * Recommendation for 'spark.sql.files.maxPartitionBytes' based on input size for each task.
+   */
+  private def recommendMaxPartitionBytes(): Unit = {
+    val res =
+      getPropertyValue("spark.sql.files.maxPartitionBytes") match {
+        case None =>
+          if (appInfo.isDefined) {
+            calculateMaxPartitionBytes(MAX_PARTITION_BYTES)
+          } else {
+            convertToMB(MAX_PARTITION_BYTES).toString
+          }
+        case Some(maxPartitionBytes) =>
+          calculateMaxPartitionBytes(maxPartitionBytes)
+      }
+    appendRecommendationForMemoryMB("spark.sql.files.maxPartitionBytes", res)
+  }
+
+  private def recommendShufflePartitions(): Unit = {
+    var shufflePartitions =
+      getPropertyValue("spark.sql.shuffle.partitions").getOrElse(DEF_SHUFFLE_PARTITIONS).toInt
+    // TODO: Need to look at other metrics for GPU spills (DEBUG mode), and batch sizes metric
+    if (appInfo.isDefined) {
+      val totalSpilledMetrics = appInfo.get.sqlTaskAggMetrics.map {
+        task => task.diskBytesSpilledSum + task.memoryBytesSpilledSum
+      }.sum
+      if (totalSpilledMetrics > 0) {
+        shufflePartitions *= DEF_SHUFFLE_PARTITION_MULTIPLIER
+        // Could be memory instead of partitions
+        appendComment("'spark.sql.shuffle.partitions' should be increased since spilling occurred.")
+      }
+      appendRecommendation("spark.sql.shuffle.partitions", s"$shufflePartitions")
+    }
+  }
+
+  def appendComment(comment: String): Unit = {
+    comments :+= comment
+  }
+
+  def convertClusterPropsToString(): String = {
+    clusterProps.toString
+  }
+
+  private def toCommentProfileResult(comments:Seq[String]): Seq[RecommendedCommentResult] = {
+    comments.map(RecommendedCommentResult)
+  }
+
+  private def toRecommendationsProfileResult: Seq[RecommendedPropertyResult] = {
+    val finalRecommendations =
+      if (FILTER_RECOMMENDATIONS) {
```

Comment:
reason we have this and its always true?

Reply:
Fixed. I moved that as an argument to `AutoTuner.getRecommendations()`